eta/core/datasets.py

```diff
@@ -1372,18 +1375,336 @@ def transform(self, src):
+                for attr in frame.attrs:
+                    if attr.name == self.attr_name:
+                        helper[1].add(attr.value)
+                        break
+
+            if len(helper[1]):
+                helper = (helper[0], list(helper[1]))
+                helper_list.append(helper)
+
+        return helper_list
+
+    def _to_helper_list_video_objects(self, records):
+        '''Balancer._to_helper_list for object attributes in videos'''
+        helper_list = []
+
+        for i, record in enumerate(records):
+            labels = record.get_labels()
+            NO_ID = 'NO_ID'
+            helper_dict = {}
+
+            for frame_no, frame in labels.frames.items():
+                if (frame_no < record.clip_start_frame
+                        or frame_no >= record.clip_end_frame):
+                    continue
+
+                for detected_object in frame.objects:
+                    if detected_object.label != self.object_label:
+                        continue
+
+                    for attr in detected_object.attrs:
+                        if attr.name == self.attr_name:
+                            obj_idx = (
+                                detected_object.index
+                                if detected_object.index is not None
+                                else NO_ID
+                            )
+
+                            if obj_idx in helper_dict:
+                                helper_dict[obj_idx].add(attr.value)
+                            else:
+                                helper_dict[obj_idx] = set([attr.value])
+
+                            break
+
+            # At this point, the keys of helper dict are unique
+            # object indices for objects of type self.object_label.
+            # The values are unique attribute values for self.attr_name.
+
+            if len(helper_dict):
+                helper = (i, [])
+                for s in helper_dict.values():
+                    helper[1].extend(list(s))
+                helper_list.append(helper)
+
+        return helper_list
+
+    def _get_target_remove(self, helper_list):
+        '''
+
+        helper_list: return value from Balancer._to_helper_list(...)
+        return: a dictionary with:
+            keys: the values of the attribute to balance
+            values: the target count to remove of this attribute value
+        '''
+        # compute counts for each attribute value
+        counts = {}
+        for idx, attr_values in helper_list:
+            for attr_value in attr_values:
+                if attr_value not in counts:
+                    counts[attr_value] = 1
+                else:
+                    counts[attr_value] += 1
+
+        # sort by count
+        counts_list = list(counts.items())
+        counts_list.sort(key=lambda x: x[1])
+
+        # integer target value
+        target_count = counts_list[int(len(counts_list) * self.median_point)][1]
```

Comment:
You can use `numpy.quantile` ([link](https://docs.scipy.org/doc/numpy/reference/generated/numpy.quantile.html)), if you first put the values into a numpy array.  This will save you having to worry about things like ties.